test/test_configuration_space.py

```diff
@@ -806,6 +808,124 @@ def test_acts_as_mapping(self):
         assert list(d.items()) == list(zip(names, hyperparameters))
         assert len(d) == 5
 
+    def test_remove_hyperparameter_priors(self):
+        cs = ConfigurationSpace()
+        integer = UniformIntegerHyperparameter('integer', 1, 5, log=True)
+        cat = CategoricalHyperparameter('cat', [0, 1, 2], weights=[1, 2, 3])
+        beta = BetaFloatHyperparameter("beta", alpha=8, beta=2, lower=-1, upper=11)
+        norm = NormalIntegerHyperparameter("norm", mu=5, sigma=4, lower=1, upper=15)
+        cs.add_hyperparameters([integer, cat, beta, norm])
+        cat_default = cat.default_value
+        norm_default = norm.default_value
+        beta_default = beta.default_value
+
+        # add some conditions, to test that remove_parameter_priors keeps the forbiddens
+        cond_1 = EqualsCondition(norm, cat, 2)
+        cond_2 = OrConjunction(EqualsCondition(beta, cat, 0),
+                               EqualsCondition(beta, cat, 1))
+        cond_3 = OrConjunction(EqualsCondition(norm, integer, 1),
+                               EqualsCondition(norm, integer, 3),
+                               EqualsCondition(norm, integer, 5))
+        cs.add_conditions([cond_1, cond_2, cond_3])
+
+        # add some forbidden clauses too, to test that remove_parameter_priors keeps the forbiddens
+        forbidden_clause_a = ForbiddenEqualsClause(cat, 0)
+        forbidden_clause_c = ForbiddenEqualsClause(integer, 3)
+        forbidden_clause_d = ForbiddenAndConjunction(forbidden_clause_a, forbidden_clause_c)
+        cs.add_forbidden_clauses([forbidden_clause_c, forbidden_clause_d])
+        uniform_cs = cs.remove_hyperparameter_priors()
+
+        expected_cs = ConfigurationSpace()
+        unif_integer = UniformIntegerHyperparameter('integer', 1, 5, log=True)
+        unif_cat = CategoricalHyperparameter('cat', [0, 1, 2], default_value=cat_default)
+
+        # These default values are computed closed form for beta (see Wikipedia), trivial for normal
```

Comment:
This comment isn't needed anymore since there's no calculation taking place

Reply:
done